### ISSUE TITLE
Row grouping

### ISSUE THREAD

**Issue by vadimffe** [w-ahmad/WinUI.TableView#69]
**Is your feature request related to a problem? Please describe.**
I am testing TableView but having hard times figuring out how to implement data grouping. Is there some example or solution hot to group data in TableView? Like in https://learn.microsoft.com/en-us/windows/communitytoolkit/controls/datagrid_guidance/group_sort_filter 

Would be good to have such a functionality.

**Comment by w-ahmad** [comment]
TableView is inherited from ListView and I don't think ListView offers item groups that can expand or collapse. However, I can try to implement item groups like ListView provides.
![image](https://github.com/user-attachments/assets/c71158c4-ee2a-43eb-a553-af44d1e44807)

**Comment by xperiandri** [comment]
Probably that should be a separate control inherited from TreeView

**Comment by cricketthomas** [comment]
Is this possible with current implementation? Using the example right from the winui gallery app and i can render the groupings keys but thats about it. 

![Image](https://github.com/user-attachments/assets/5f7744e2-f6a5-4d6b-878b-a9008abc27c3)

```
  <tv:TableView.GroupStyle>
      <GroupStyle>
          <GroupStyle.HeaderTemplate>
              <DataTemplate x:DataType="models:GroupInfoList">
                  <Border AutomationProperties.AccessibilityView="Raw">
                      <TextBlock
                          AutomationProperties.AccessibilityView="Raw"
                          Style="{ThemeResource TitleTextBlockStyle}"
                          Text="{x:Bind Key}" />
                  </Border>
              </DataTemplate>
          </GroupStyle.HeaderTemplate>
      </GroupStyle>
  </tv:TableView.GroupStyle>
```

**Comment by YakubMordon** [comment]
Hi everyone, how's the progress on this thread?

Is grouping ready? If not, I'm willing to contribute code to implement this feature, because I plan to use Grouping functionality in my Project with TableView as well

**Comment by w-ahmad** [comment]
@YakubMordon check out PR #425 and let me know what you think. I’d really appreciate your feedback on the implementation and any suggestions you may have.

**Comment by YakubMordon** [comment]
Yep, this is what I really need, overall code looks very good, thank you very much for supporting this library😊
